streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/DefaultApplicationState.java

```diff
@@ -32,22 +33,29 @@
 import org.apache.kafka.streams.processor.assignment.ProcessId;
 import org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor;
 
-public class ApplicationStateImpl implements ApplicationState {
+public class DefaultApplicationState implements ApplicationState {
 
     private final AssignmentConfigs assignmentConfigs;
-    private final Set<TaskInfo> tasks;
+    private final Map<TaskId, TaskInfo> tasks;
     private final Map<UUID, ClientMetadata> clientStates;
 
-    public ApplicationStateImpl(final AssignmentConfigs assignmentConfigs,
-                                final Set<TaskInfo> tasks,
-                                final Map<UUID, ClientMetadata> clientStates) {
+    private final Map<Boolean, Map<ProcessId, KafkaStreamsState>> cachedKafkaStreamStates;
+
+    public DefaultApplicationState(final AssignmentConfigs assignmentConfigs,
+                                   final Set<TaskInfo> tasks,
+                                   final Map<UUID, ClientMetadata> clientStates) {
         this.assignmentConfigs = assignmentConfigs;
-        this.tasks = unmodifiableSet(tasks);
+        this.tasks = unmodifiableMap(tasks.stream().collect(Collectors.toMap(TaskInfo::id, task -> task)));
```

Comment:
We can avoid adding yet another `.stream().collect() ` by building up the map with the TaskIds as keys when we're building the `Set<TaskInfo> tasks` parameter in the caller of this constructor.